src/engine/Marabou.cpp

```diff
@@ -30,33 +30,23 @@
 #undef ERROR
 #endif
 
-Marabou::Marabou()
-    : _acasParser( NULL )
-    , _onnxParser( NULL )
+Marabou::Marabou( InputQuery& inputQuery )
+    : _inputQuery( inputQuery )
     , _engine()
 {
 }
 
 Marabou::~Marabou()
 {
-    if ( _acasParser )
-    {
-        delete _acasParser;
-        _acasParser = NULL;
-    }
-
-    if ( _onnxParser )
-    {
-        delete _onnxParser;
-        _onnxParser = NULL;
-    }
 }
 
 void Marabou::run()
 {
+    if ( Options::get()->getBool( Options::DEBUG_ASSIGNMENT ) )
+        importDebuggingSolution();
+
     struct timespec start = TimeUtils::sampleMicro();
 
-    prepareInputQuery();
```

Comment:
So the one change that I can see is that parsing the input query is no longer timed. If that is important I can add a separate timing function to time it?

Reply:
I think this is fine for now. Maybe add a TODO in the comment so we can revisit it later?